api_design/DDS-1061-optimized_directory_structure_endpoint.md

```diff
@@ -12,143 +12,67 @@ The goal of this development epic is to produce a set of endpoints to be used in
 
 #### Background
 
-The legacy children endpoints have proven to be suboptimal for the GCB clients needs. The GCB client needs a paginated endpoint to list ALL children of a project or folder. The current children endpoints meet the information needs for their use, but it currently takes on average 15 minutes to get the directory structure for a project or folder with many subfolders and files (similar to our ua_test sprawl project).
+The legacy children endpoints have proven to be suboptimal for the GCB clients needs. The GCB client needs an endpoint or set of endpoints to facilitate uploading and downloading of multiple files, usually entire projects. The current children endpoints meet the information needs for their use, but it currently takes on average 15 minutes to get the directory structure for a project or folder with many subfolders and files (similar to our ua_test sprawl project).
 
 #### Proposal
 
 In this feature epic we will engage the GCB dds_client development team for feedback on two areas:
 
-- determine minimal query parameters required for their needs, for example, this endpoint may not need to support `name_contains` or `exclude_response_fields`
+- determine minimal query parameters required for their needs
 - designing a response document with the minimal fields required to meet the dds_client requirements
 - include deleted children in response documents to facilitate detection of changes in the overall filesystem
 - sort children in order of last_updated_on to ensure most recently changed objects come out first in responses
 
 #### Document Design
 
-We will start with the following exemplar of the response provided by the current children endpoint. The goal of this phase is to tease out the specific information in this response, and craft a new JSON structure with these specific pieces of information only.
+We started with an exemplar of the response provided by the current children endpoint. The goal of this phase was to tease out the specific information in this response, and craft a new JSON structure with these specific pieces of information only. This was used to create a proposed Summary of impacted APIs and API Specification (below).
+
+## Implementation View
+
+#### Summary of impacted APIs
+
+|Endpoint |Description |
+|---|---|
+| `GET /projects/:id/files` | Get list of files in a project. |
+| `GET /folders/:id/files` | Get list of files in a folder. |
```

Comment:
Let's focus on `GET /projects/:id/files` and lose the `GET /folders/:id/files` for now.